android/src/main/java/com/reactcommunity/rndatetimepicker/CustomTimePickerDialog.java

```diff
@@ -0,0 +1,220 @@
+      return minute * mTimePickerInterval;
+    }
+
+    return minute;
+  }
+
+  /**
+   * 'Snaps' real minutes to nearest valid value
+   *
+   * @param realMinutes 'real' minutes (0-59)
+   * @return nearest valid value
+   */
+  private int snapMinutesToInterval(int realMinutes) {
+    float stepsInMinutes = (float) realMinutes / (float) mTimePickerInterval;
+
+    if (mDisplay == RNTimePickerDisplay.SPINNER) {
+      return Math.round(stepsInMinutes);
+    }
+
+    return Math.round(stepsInMinutes) * mTimePickerInterval;
+  }
+
+  /**
+   * Determines if real minutes align setted minuteInterval
+   *
+   * @param realMinutes 'real' minutes (0-59)
+   */
+  private boolean minutesAreInvalid(int realMinutes) {
+    return realMinutes % mTimePickerInterval != 0;
+  }
+
+  /**
+   * Determines if the picker is in text input mode (keyboard icon in 'clock' mode)
+   */
+  private boolean pickerIsInTextInputMode() {
+    int textInputPickerId = mContext.getResources().getIdentifier("input_mode", "id", "android");
+    final View textInputPicker = this.findViewById(textInputPickerId);
+
+    return textInputPicker != null && textInputPicker.hasFocus();
+  }
+
+  /**
+   * Corrects minute values if they don't align with minuteInterval
+   *
+   * If the picker is in text input mode, correction will be postponed slightly to let the user
+   * finish the input
+   *
+   * If the picker is not in text input mode, correction will take place immidiately
+   *
+   * @param view the picker's view
+   * @param hourOfDay the picker's selected hours
+   * @param realMinutes 'real' minutes (0-59)
+   */
+  private void correctEnteredMinutes(final TimePicker view, final int hourOfDay, final int realMinutes) {
+    if (pickerIsInTextInputMode()) {
+      final EditText textInput = (EditText) view.findFocus();
+
+      // 'correction' callback
+      runnable = new Runnable() {
+        @Override
+        public void run() {
+          // set valid minutes && move caret to the end of input
+          view.setCurrentMinute(snapMinutesToInterval(realMinutes));
+          view.setCurrentHour(hourOfDay);
+          textInput.setSelection(textInput.getText().length());
+        }
+      };
+
+      handler.postDelayed(runnable, 500);
+      return;
+    }
+
+    view.setCurrentMinute(snapMinutesToInterval(realMinutes));
+    view.setCurrentHour(hourOfDay);
+  }
+
+  @Override
+  public void onTimeChanged(final TimePicker view, final int hourOfDay, final int minute) {
+    final int realMinutes = getRealMinutes(minute);
+
+    // remove pending 'validation' callbacks if any
+    handler.removeCallbacks(runnable);
+
+    if (minutesAreInvalid(realMinutes)) {
+      correctEnteredMinutes(view, hourOfDay, realMinutes);
+      return;
+    }
+
+    super.onTimeChanged(view, hourOfDay, realMinutes);
+  }
+
+  @Override
+  public void onClick(DialogInterface dialog, int which) {
+    switch (which) {
+      case BUTTON_POSITIVE:
```

Comment:
I'd find it preferable to handle only the event we're interested in (BUTTON_POSITIVE) when we're interested in it (when user sets custom minuteinterval) and defer handling other siturations to superclass, WDYT?

Reply:
Yes, I agree I'll add a condition to check if minute interval is not set and call the `onClick` from `superclass`